clang/lib/DPCT/ExprAnalysis.cpp

```diff
@@ -587,6 +587,28 @@ void ExprAnalysis::analyzeExpr(const IntegerLiteral *IL) {
   }
 }
 
+void ExprAnalysis::analyzeExpr(const InitListExpr *ILE) {
+  if (const CXXFunctionalCastExpr *CFCE =
+          DpctGlobalInfo::findParent<CXXFunctionalCastExpr>(ILE)) {
+    if (CFCE->isListInitialization() && CFCE->getType()->isIntegerType() &&
+        ILE->getNumInits() == 1 &&
+        !ILE->getInit(0)->isIntegerConstantExpr(Context)) {
+      // Replace initializer list with explicit type conversion (e.g.,
+      // 'int64_t{d3[2]}' to 'int64_t(d3[2])') to slience narrowing error (e.g.,
```

Comment:
An example to such error in https://godbolt.org/z/M3rcMdd9v
` error: non-constant-expression cannot be narrowed from type 'size_t' (aka 'unsigned long') to 'int64_t' (aka 'long') in initializer list [-Wc++11-narrowing]`